python/across_api/auth/api.py

```diff
@@ -0,0 +1,90 @@
+# Copyright © 2023 United States Government as represented by the
+# Administrator of the National Aeronautics and Space Administration.
+# All Rights Reserved.
+
+
+import os
+from typing import Optional
+
+import jwt
+import requests
+from fastapi import Depends, HTTPException, Request, status
+from fastapi.security import HTTPAuthorizationCredentials, HTTPBearer
+
+from ..base.api import app
+from .schema import VerifyAuth
+
+
+class JWTBearer(HTTPBearer):
+    token_endpoint: str
+    jwks_uri: str
+
+    def __init__(self, **kwargs):
+        # Configure URL from IdP
+        user_pool_id = os.environ.get("COGNITO_USER_POOL_ID")
+        if user_pool_id is not None:
+            provider_url = f"https://cognito-idp.{user_pool_id.split('_')[0]}.amazonaws.com/{user_pool_id}/"
+        elif os.environ.get("ARC_ENV") == "testing":
+            provider_url = f"http://localhost:{os.environ.get('ARC_OIDC_IDP_PORT')}/"
+        else:
+            raise HTTPException(
+                status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+                detail="Environment variable COGNITO_USER_POOL_ID must be defined in production.",
+            )
```

Comment:
We don't need to specifically generate HTTP 500 errors because uncaught exceptions of _any_ kind other than HTTPException itself will be reported as HTTP 500 responses.

```suggestion
            raise RuntimeError(
                "Environment variable COGNITO_USER_POOL_ID must be defined in production.",
            )
```